doc/index.rst

```diff
@@ -207,26 +225,76 @@ Pulling Docker image from an online hub
 
 You can pull any `Docker` image by specifying in `DOCKER_IMAGE` variable, as long as the following requirement is met:
 
-* `python-catkin-tools`, `python-pip`, `python-rosdep`, `python-wstool`
 * sources.list set up (`example <http://wiki.ros.org/kinetic/Installation/Ubuntu#Installation.2BAC8-Ubuntu.2BAC8-Sources.Setup_your_sources.list>`_).
+* `python-catkin-tools`, `python-pip`, `python-rosdep`, `python-wstool`. If your Docker image is missing any of the above libraries, then you can still pass their name by `ADDITIONAL_DEBS` (see `variables section <./index.rst#optional-environment-variables>`_).
+
+Some more notes:
+
+* Setting `DOCKER_IMAGE` is a bit tricky:
+   * disables the set-up of ROS based on `ROS_REPO` (or non-recommended `ROS_REPOSITORY_PATH`), and ROS_DISTRO.
+   * but `ROS_DISTRO` still needs to be set.  
+* Some common credentials such as `.docker`, `.ssh`, `.subversion` are passed from CI native platform to Docker container.
+
+Pass custom variables to Docker
+-------------------------------
+
+On CI platform usually some variables are available for the convenience. Since all checks using `industrial_ci` are NOT running directly on the operating system running on CI, but instead running on `Docker` where those variables are not defined, dozens of them are already passed for you (you can see `the list of those variables <https://github.com/ros-industrial/industrial_ci/blob/master/industrial_ci/src/docker.env>`_).
+
+Still, you may want to pass some other vars. `DOCKER_RUN_OPTS='-e MY_VARIABLE_VALUE'` should do the trick.
+You can even set it to a specific value: `DOCKER_RUN_OPTS='-e MY_VARIABLE_VALUE=42'` (format varies per CI platform. These are Gitlab CI example).
+
+(Gitlab CI) Access to private repositories
+------------------------------------------
 
-If your Docker image is missing any of the above libraries, then you can still pass their name by `ADDITIONAL_DEBS` (see `variables section <./index.rst#optional-environment-variables>`_).
+To open PRs/MRs (merge requests, that's what Gitlab calls pull requests) that access private Gitlab repos, additional setting is needed both on:
```

Comment:
Instructions should not be limited to opening PRs/MRs. Any repo using GitLab CI where `industrial_ci` needs to access private repositories (anywhere), needs to setup authentication to do so.